Add Sanitizers documentation

uilianries opened the pull request:
Related to https://github.com/conan-io/examples2/pull/192

Also related to Conan issues:

* https://github.com/conan-io/conan/issues/13981
* https://github.com/conan-io/conan/issues/17797
* https://github.com/conan-io/conan/issues/15374

czoido commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
```diff
@@ -0,0 +1,356 @@
+### Compiler sanitizers
+
```
```suggestion
```
Is that line there for a reason?

czoido commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
```diff
@@ -0,0 +1,354 @@
+
+.. warning::
+
+   Using sanitizers in production, particularly with SUID binaries, is dangerous. The libsanitizer
+   runtimes rely on environment variables that could enable privilege escalation attacks.
+   Use sanitizers only in development and testing environments.
```
I think something like this would read better:

```
Using sanitizers in production, especially with programs that run with elevated privileges (for example, SUID binaries on Linux), is dangerous.
The sanitizer runtime libraries depend on environment variables, which could allow privilege escalation attacks.
Use sanitizers only in development and testing environments.
```

czoido commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
I think it would be good, before this section, to add a bit of explanation about how sanitizers influence binary compatibility and then, as a result, explain how you can model it for these cases.

Also, it would be great to do some research about if all the sanitizers will make binaries to break compatibility or there are cases were you can apply the sanitizers just as flags without the risk of producing incompatible binaries.

czoido commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
I personally don't like this way of adding the sanitizers "in the shadows". I think it's much better to do this explicitly in profiles, so I think I would not document this.

uilianries replied:
Agreed. This is documented in Conan 1.x as well, but indeed keeping it tracked in the profiles only is much safer.

uilianries commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
Just added some tables, indeed it is much clearer and visible now. Thank you for that suggestion! Please, take a look.

uilianries commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
Conan export + conan install is olde school. Let me update to `conan build` then :)

uilianries replied:
Done, updated in the commit 81e012b8

uilianries commented on examples/dev_flow/sanitizers/compiler_sanitizers.rst:
Added an explanation on the commit 4311346

memsharded commented on security/sanitizers.rst:
The ``conan export`` is necessary? I think it is not.

uilianries replied:
Simplified by commit 5786822ac616f6c1069577dbe01b3fd9874d38d9

memsharded commented on security/sanitizers.rst:
Same as above

czoido commented on security/sanitizers.rst:
For this one, checking the docs looks like it's supported for both x86 and x64 on Windows 10 and later: https://learn.microsoft.com/en-us/cpp/sanitizers/asan?view=msvc-170

czoido commented on security/sanitizers.rst:
I think all the example code would be better in the examples section, and linked from this page.

czoido commented on security/sanitizers.rst:
Yes, makes sense, thanks!

uilianries commented on security/sanitizers.rst:
Improved by commit 6a055589. Let me know if is too much, or we just should avoid this section in docs.

uilianries commented:
@czoido @memsharded Hello! Thank you for another round of review. I just addressed your comments in my last commits. You can take a quick preview of how the documentation is on the commit [74acf05](https://github.com/conan-io/docs/pull/4199/commits/74acf05546e20559d6ab441bf3ece93f2a24a058) here:

<img width="1920" height="8051" alt="Screenshot 2025-11-05 at 12-27-03 Sanitizers — conan 2 21 0 documentation" src="https://github.com/user-attachments/assets/071983be-2c66-48fd-8e94-f645eef2c069" />

<img width="1920" height="3324" alt="Screenshot 2025-11-05 at 12-26-36 Building Examples Using Sanitizers — conan 2 21 0 documentation" src="https://github.com/user-attachments/assets/59dfb764-10a0-4464-9a30-a871c3296d75" />

memsharded commented on security/sanitizers.rst:
```diff
@@ -0,0 +1,352 @@
+.. _security_sanitizers:
+
+Sanitizers
```
For discoverability, search purposes, something like ``C, C++ compiler sanitizers`` or something in that line, a bit more complete, could be nice

memsharded commented on security/sanitizers.rst:
This can be a bit confusing/not reproducible? Is this a gcc based or a clang profile?
Because the ``settings_user.yml`` above only added the sanitizer model to ``clang``, not ``gcc``, but the default profile in Linux systems is generally ``gcc``.

Maybe it is better to fully explicitly define the profile? Or if not, maybe a comment saying the default profile is a clang one?

uilianries replied:
Fixed by 0779b86452f9fd6e45bc5ed971e1d388633276a3

memsharded commented on security/sanitizers.rst:
Same here, at least a note that this is ``compiler=msvc`` and not others.

memsharded commented on security/sanitizers.rst:
Maybe the section can be kept, but it would be necessary to at least clearly clarify that it only works when all dependencies are using CMake as build system and ``CMakeToolchain`` integration, it will not work for dependencies with Meson, autotools or others.

uilianries commented on examples/security/sanitizers.rst:
```diff
@@ -0,0 +1,9 @@
+.. toctree::
+   :maxdepth: 2
+
+   sanitizers/compiler_sanitizers
```
Moved to sanitizer folder by the commit d596bc210c518d2c3d82a8fc962d4ae6513b9318

memsharded reviewed (approved):
Good for me, just a note about ``clang`` compiler profiles in Linux
